pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderToken.java

```diff
@@ -99,7 +100,8 @@
 
     @Override
     public void close() throws IOException {
-        // noop
+        expiredTokenMetrics.clear();
+        expiringTokenMinutesMetrics.clear();
```

Comment:
The `expiredTokenMetrics` is static and therefore might be shared across multiple instances. Calling clear on one instance close is probably not ok.

Reply:
Well the static metrics cause some of the tests to flake, this fixed the test issues, for running services I think clearing static metrics during close makes sense.

Reply:
That seems to make the prod code to bend around the test structure. 

In general, it's problematic to rely on the metrics values, precisely because they're static and if multiple tests are running in parallel that will break all assumptions anyway. 

For this case, if we *really* need to reset the stats, it could instead be done through a static `resetStats()` methods that is package protected, instead of relying on the `close()`.

Reply:
ok will add a package level reset method.

Reply:
done